### ISSUE TITLE
Incorrect header in wav files can lead to enormous memory allocations (+4gb)

### ISSUE THREAD

**Issue by ChichiMsdk** [libsdl-org/SDL#10052]
When loading a wav file using `SDL_LoadWAV(fname, &spec, &audio_buf, &wav_length)`, an incorrect header that reports an arbitrary amount of data could lead to several gb of allocations performed on the `audio_buf` <br>Despite knowing the correct size of the audio file since it is correctly returned by `wav_length` <br> 
In my case the file was around 70mb and reported it's size of -1 and therefore SDL allocated the **signed int max 32bits** value (+4gb !)
I guess we should instead allocate the correct amount by checking the header against the size calculated with `wav_length`

**Comment by icculus** [comment]
Do you happen to have a .wav file that triggers this that you can upload here, so I'm definitely fixing the right problem?

**Comment by ChichiMsdk** [comment]
Yes but github won't allow me to upload .wav nor files > 10mb so here's a wetransfer link:
https://we.tl/t-wUrId8xBfx
Screenshot for more explanation of what I meant
<img width="865" alt="Screenshot 2024-06-18 204942" src="https://github.com/libsdl-org/SDL/assets/111235918/5054f583-3d46-4ac9-90e2-8f8bfa2a621d">

**Comment by icculus** [comment]
Thanks! I'll grab this later today!

**Comment by slouken** [comment]
@ChichiMsdk, can you repost the link? It's expired here.

**Comment by ChichiMsdk** [comment]
@slouken I don't have the file anymore but this could be easily reproduced by manipulating the 4 bytes after "data" in the wav header for any wav files. Also I grabbed some youtube video that I converted directly using this online tool [https://cobalt.tools/](url).

I believe that the wrong information from the header comes from this tool
